### ISSUE TITLE
ContextErrorException while hide the only order state associate to an order

### ISSUE THREAD

**Issue by prestamodule** [PrestaShop/PrestaShop#13838]
**Describe the bug**
If you edit an Order State to be hidden inside the client account and that it's the only one associated to the order, the getCustomerOrders() thrown an error.

**To Reproduce**
1. Go to Back Office and edit an order state (Payment by check, for demo datas)
2. Disable the "Show to customer" option
3. Save
4. Go to the customer account and check it's history

**Screenshots**
![image](https://user-images.githubusercontent.com/1162527/57927957-c9195000-78af-11e9-8351-9a6614189b27.png)

**Additionnal information**
PrestaShop version: 1.7.6.0-beta1
PHP version: N/A

**Comment by khouloudbelguith** [comment]
Hi @prestamodule,

Steps to reproduce the issue:
Go to BO => Shop Parameters => Order Settings => Statuses page
Edit the Payment by check status
Check "Hide this status in all customer orders." option
Go to FO => order history of John Doe

Thanks!

**Comment by khouloudbelguith** [comment]
@prestamodule, I manage to reproduce the issue with PS1.7.5.2  & PS1.7.6.0beta1
![image](https://user-images.githubusercontent.com/16067358/57929065-3de98b80-78aa-11e9-9be9-c7eff4c83da8.png)

I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by rmilecki** [comment]
I can reproduce this exception using 1.7.6.2 release with "Debug mode" enabled. Line number is just a bit different there:
```
in Order.php line 944
```
That points to the following code:
```
$ cat -n ./classes/order/Order.php | egrep -B 1 -A 3 "^ *944"
   943          foreach ($res as $key => $val) {
   944              $res[$key]['order_state'] = $indexedOrderStates[$val['id_order_state']]['name'];
   945              $res[$key]['invoice'] = $indexedOrderStates[$val['id_order_state']]['invoice'];
   946              $res[$key]['order_state_color'] = $indexedOrderStates[$val['id_order_state']]['color'];
   947          }
```
"Debug mode" is required in my case as it seems to enable `E_NOTICE` errors reporting.

The problem is that `$val['id_order_state']` is `NULL`. It results in above code looking like:
```
$indexedOrderStates[NULL]['name'];
$indexedOrderStates[NULL]['invoice'];
$indexedOrderStates[NULL]['color']
```
which causes `Notice:  Undefined index:` of course.

**Comment by rmilecki** [comment]
This bug has been partially fixed by the commit 9c9dac676283 ("Fix DuplicateOrderCartHandler context needs customer, add behat tests, improve exception message display in dev mode"). That change adds:
```
$orderState = !empty($val['id_order_state']) ? $indexedOrderStates[$val['id_order_state']] : null;
```
which may result in slightly better:
```
null['name'];
null['invoice'];
null['color'];
```
That works well with PHP up to 7.3 but not with PHP 7.4. It's because PHP 7.4 doesn't allow things like `null['foo']` anymore:
```
Notice:  Trying to access array offset on value of type null
```

**Comment by colinegin** [comment]
ping @PierreRambaud as you are working on compatibility with PHP 7.4 ;)
I have put the issue in the 178 kanban

**Comment by LouiseBonnard** [comment]
Hi @prestamodule, can you confirm that PR #17337 solves your issue?

**Comment by PierreRambaud** [comment]
@colinegin if this issue is related to PHP 7.4, it's a duplicate of #16477

**Comment by PierreRambaud** [comment]
I confirm it's a duplicate of #16477 and it's fixed in #18787

**Comment by LouiseBonnard** [comment]
Hey @prestamodule, thanks for your report. Since this is a duplicate of #16477 (soon-to-be fixed by #18787), I close this issue so that we can all focus on one issue instead of multiple duplicates. Feel free to follow up on #16477, or reopen this one if need be. :-)
